library/fluid/fluid_extensions/src/main/java/org/quiltmc/qsl/fluid/api/QuiltFluid.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.fluid.api;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.block.entity.BlockEntity;
+import net.minecraft.fluid.FlowableFluid;
+import net.minecraft.fluid.Fluid;
+import net.minecraft.fluid.FluidState;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Direction;
+import net.minecraft.world.BlockView;
+import net.minecraft.world.WorldAccess;
+import net.minecraft.world.WorldView;
+
+public abstract class QuiltFluid extends FlowableFluid implements FlowableFluidExtensions {
+	/**
+	 * @return whether the given fluid an instance of this fluid
+	 */
+	@Override
+	public boolean matchesType(Fluid fluid) {
+		return fluid == getStill() || fluid == getFlowing();
+	}
+
+	/**
+	 * @return whether the fluid infinite like water
+	 */
+	@Override
+	protected boolean isInfinite() {
+		return false;
```

Comment:
this could be tag-driven maybe

Reply:
doing it in a later pr. This pr shall only provide the neccesary things to work on the tag-driven / data-drive things later on. It is intended tho.